spot_driver/src/images/spot_image_publisher.cpp

```diff
@@ -125,5 +132,6 @@ void SpotImagePublisher::timerCallback(bool uncompress_images, bool publish_comp
   middleware_handle_->publishCompressedImages(image_result.value().compressed_images_);
 
   tf_broadcaster_->updateStaticTransforms(image_result.value().transforms_);
+  logger_->logError(std::string("timer callback[") + std::to_string(local_count) + "] end.");
```

Comment:
That's the resulting output on my end:
_image_publisher_node-2] [ERROR] [1714726714.229783612] [spot_image_publisher]: timer callback[978] end.
[spot_image_publisher_node-2] [ERROR] [1714726714.229875451] [spot_image_publisher]: timer callback[1010] start.
[spot_image_publisher_node-2] [ERROR] [1714726714.728457919] [spot_image_publisher]: timer callback[979] end.
[spot_image_publisher_node-2] [ERROR] [1714726714.728727863] [spot_image_publisher]: timer callback[1011] start.
[spot_image_publisher_node-2] [ERROR] [1714726715.196232866] [spot_image_publisher]: timer callback[980] end.
[spot_image_publisher_node-2] [ERROR] [1714726715.196359714] [spot_image_publisher]: timer callback[1012] start.
[spot_image_publisher_node-2] [ERROR] [1714726715.759977800] [spot_image_publisher]: timer callback[981] end.
[spot_image_publisher_node-2] [ERROR] [1714726715.760243804] [spot_image_publisher]: timer callback[1013] start.
[spot_image_publisher_node-2] [ERROR] [1714726716.226881856] [spot_image_publisher]: timer callback[982] end.
[spot_image_publisher_node-2] [ERROR] [1714726716.227183485] [spot_image_publisher]: timer callback[1014] start.
[spot_image_publisher_node-2] [ERROR] [1714726716.792703279] [spot_image_publisher]: timer callback[983] end.
[spot_image_publisher_node-2] [ERROR] [1714726716.792922622] [spot_image_publisher]: timer callback[1015] start.
[spot_image_publisher_node-2] [ERROR] [1714726717.821144610] [spot_image_publisher]: timer callback[984] end.
[spot_image_publisher_node-2] [ERROR] [1714726717.821356942] [spot_image_publisher]: timer callback[1016] start.
[spot_image_publisher_node-2] [ERROR] [1714726718.993255965] [spot_image_publisher]: timer callback[985] end.
[spot_image_publisher_node-2] [ERROR] [1714726718.993474957] [spot_image_publisher]: timer callback[1017] start.
[spot_image_publisher_node-2] [ERROR] [1714726719.867897860] [spot_image_publisher]: timer callback[986] end.
[spot_image_publisher_node-2] [ERROR] [1714726719.868095185] [spot_image_publisher]: timer callback[1018] start.
[spot_image_publisher_node-2] [ERROR] [1714726720.798031487] [spot_image_publisher]: timer callback[987] end.
[spot_image_publisher_node-2] [ERROR] [1714726720.798229871] [spot_image_publisher]: timer callback[1019] start.
[spot_image_publisher_node-2] [ERROR] [1714726721.267579114] [spot_image_publisher]: timer callback[988] end.
[spot_image_publisher_node-2] [ERROR] [1714726721.267790080] [spot_image_publisher]: timer callback[1020] start.
[spot_image_publisher_node-2] [ERROR] [1714726721.907005275] [spot_image_publisher]: timer callback[989] end.
[spot_image_publisher_node-2] [ERROR] [1714726721.907205510] [spot_image_publisher]: timer callback[1021] start.
[spot_image_publisher_node-2] [ERROR] [1714726722.416965921] [spot_image_publisher]: timer callback[990] end.
[spot_image_publisher_node-2] [ERROR] [1714726722.417147082] [spot_image_publisher]: timer callback[1022] start.
[spot_image_publisher_node-2] [ERROR] [1714726723.165529209] [spot_image_publisher]: timer callback[991] end.
[spot_image_publisher_node-2] [ERROR] [1714726723.165712088] [spot_image_publisher]: timer callback[1023] start.
[spot_image_publisher_node-2] [ERROR] [1714726723.777925553] [spot_image_publisher]: timer callback[992] end.
[spot_image_publisher_node-2] [ERROR] [1714726723.778156946] [spot_image_publisher]: timer callback[1024] start.
[spot_image_publisher_node-2] [ERROR] [1714726724.798056708] [spot_image_publisher]: timer callback[993] end.
[spot_image_publisher_node-2] [ERROR] [1714726724.798193917] [spot_image_publisher]: timer callback[1025] start.
[spot_image_publisher_node-2] [ERROR] [1714726725.662917009] [spot_image_publisher]: timer callback[994] end.
[spot_image_publisher_node-2] [ERROR] [1714726725.663247123] [spot_image_publisher]: timer callback[1026] start.
[spot_image_publisher_node-2] [ERROR] [1714726726.690995008] [spot_image_publisher]: timer callback[995] end.
[spot_image_publisher_node-2] [ERROR] [1714726726.691063692] [spot_image_publisher]: timer callback[1027] start.
[spot_image_publisher_node-2] [ERROR] [1714726727.513240747] [spot_image_publisher]: timer callback[996] end.
[spot_image_publisher_node-2] [ERROR] [1714726727.513488717] [spot_image_publisher]: timer callback[1028] start.
[spot_image_publisher_node-2] [ERROR] [1714726728.283715766] [spot_image_publisher]: timer callback[997] end.
[spot_image_publisher_node-2] [ERROR] [1714726728.283909614] [spot_image_publisher]: timer callback[1029] start.
[spot_image_publisher_node-2] [ERROR] [1714726728.723184760] [spot_image_publisher]: timer callback[998] end.
[spot_image_publisher_node-2] [ERROR] [1714726728.723522497] [spot_image_publisher]: timer callback[1030] start.
[spot_image_publisher_node-2] [ERROR] [1714726729.218585761] [spot_image_publisher]: timer callback[999] end.
[spot_image_publisher_no


We are in a ballpark of 500ms for the data transfer. 

Analysing a quick ros2 bag recording (see below); I get:
data-rate: 1,68 MiB / sec
frequency: 1,56 fps

The network cannot be the restricting factor here.


Files:             rosbag2_2024_05_03-11_01_17_0.db3
Bag size:          273.5 MiB
Storage id:        sqlite3
Duration:          162.766s
Start:             May  3 2024 11:01:17.378 (1714726877.378)
End:               May  3 2024 11:04:00.144 (1714727040.144)
Messages:          87598
                   Topic: /depth/frontright/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 254 | Serialization Format: cdr
                   Topic: /depth/back/image | Type: sensor_msgs/msg/Image | Count: 253 | Serialization Format: cdr
                   Topic: /depth/frontleft/image | Type: sensor_msgs/msg/Image | Count: 253 | Serialization Format: cdr
                   Topic: /status/behavior_faults | Type: spot_msgs/msg/BehaviorFaultState | Count: 6847 | Serialization Format: cdr
                   Topic: /status/battery_states | Type: spot_msgs/msg/BatteryStateArray | Count: 6847 | Serialization Format: cdr
                   Topic: /status/system_faults | Type: spot_msgs/msg/SystemFaultState | Count: 6847 | Serialization Format: cdr
                   Topic: /odometry | Type: nav_msgs/msg/Odometry | Count: 6847 | Serialization Format: cdr
                   Topic: /depth/back/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 254 | Serialization Format: cdr
                   Topic: /status/wifi | Type: spot_msgs/msg/WiFiState | Count: 6847 | Serialization Format: cdr
                   Topic: /status/power_states | Type: spot_msgs/msg/PowerState | Count: 6846 | Serialization Format: cdr
                   Topic: /tf_static | Type: tf2_msgs/msg/TFMessage | Count: 2 | Serialization Format: cdr
                   Topic: /status/leases | Type: spot_msgs/msg/LeaseArray | Count: 152 | Serialization Format: cdr
                   Topic: /status/feet | Type: spot_msgs/msg/FootStateArray | Count: 6847 | Serialization Format: cdr
                   Topic: /depth/right/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 254 | Serialization Format: cdr
                   Topic: /status/metrics | Type: spot_msgs/msg/Metrics | Count: 158 | Serialization Format: cdr
                   Topic: /depth/frontleft/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 254 | Serialization Format: cdr
                   Topic: /tf | Type: tf2_msgs/msg/TFMessage | Count: 9243 | Serialization Format: cdr
                   Topic: /status/feedback | Type: spot_msgs/msg/Feedback | Count: 3236 | Serialization Format: cdr
                   Topic: /status/mobility_params | Type: spot_msgs/msg/MobilityParams | Count: 3236 | Serialization Format: cdr
                   Topic: /depth/left/image | Type: sensor_msgs/msg/Image | Count: 253 | Serialization Format: cdr
                   Topic: /rosout | Type: rcl_interfaces/msg/Log | Count: 564 | Serialization Format: cdr
                   Topic: /depth/right/image | Type: sensor_msgs/msg/Image | Count: 254 | Serialization Format: cdr
                   Topic: /odometry/twist | Type: geometry_msgs/msg/TwistWithCovarianceStamped | Count: 6847 | Serialization Format: cdr
                   Topic: /depth/left/camera_info | Type: sensor_msgs/msg/CameraInfo | Count: 254 | Serialization Format: cdr
                   Topic: /status/estop | Type: spot_msgs/msg/EStopStateArray | Count: 6847 | Serialization Format: cdr
                   Topic: /joint_states | Type: sensor_msgs/msg/JointState | Count: 6847 | Serialization Format: cdr
                   Topic: /depth/frontright/image | Type: sensor_msgs/msg/Image | Count: 254 | Serialization Format: cdr
                   Topic: /robot_description | Type: std_msgs/msg/String | Count: 1 | Serialization Format: cdr